hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java

```diff
@@ -47,8 +48,27 @@ private EthereumTransactionParser getEthereumTransactionParser(byte[] transactio
             throw new InvalidDatasetException("Ethereum transaction bytes length is less than 2 bytes in length");
         }
 
-        var eip1559StartingBytesMatch = transactionBytes[0] == EIP1559_BYTES_PREFIX[0] &&
-                transactionBytes[1] == EIP1559_BYTES_PREFIX[1];
-        return eip1559StartingBytesMatch ? eip1559EthereumTransactionParser : legacyEthereumTransactionParser;
+        var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(
+                transactionBytes);
+        var legacyRlpItem = decoder.next();
+        Byte legacyRlpItemByte;
+
+        try {
+            legacyRlpItemByte = legacyRlpItem.asByte();
+        } catch (IllegalArgumentException illEx) {
+            // catch out of range length exception
+            return legacyEthereumTransactionParser;
+        }
+
+        if (legacyRlpItemByte != EIP1559_TYPE_BYTE) {
+            return legacyEthereumTransactionParser;
+        }
+
+        var eip1559RlpItem = decoder.next();
+        if (!eip1559RlpItem.isList()) {
+            return legacyEthereumTransactionParser;
+        }
+
+        return eip1559EthereumTransactionParser;
```

Comment:
We should use the exact same decision logic as [services](https://github.com/hashgraph/hedera-services/blob/a194837e3b7d755bde0de5c1130c86ddeb09cd2b/hapi-utils/src/main/java/com/hedera/services/ethereum/EthTxData.java#L83) uses:

```suggestion
        var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(transactionBytes);
        var rlpItem = decoder.next();
        return rlpItem.isList() ? legacyEthereumTransactionParser : eip1559EthereumTransactionParser;
```

Reply:
Simplified